crypto/signer_test.go

```diff
@@ -94,3 +96,110 @@ func TestSignLogRootSignerFails(t *testing.T) {
 	_, err = NewSHA256Signer(s).Sign(hash)
 	testonly.EnsureErrorContains(t, err, "signfail")
 }
+
+func TestSignLogRoot(t *testing.T) {
+	key, err := pem.UnmarshalPrivateKey(testonly.DemoPrivateKey, testonly.DemoPrivateKeyPass)
+	if err != nil {
+		t.Fatalf("Failed to open test key, err=%v", err)
+	}
+	signer := NewSHA256Signer(key)
+
+	for _, test := range []struct {
+		root trillian.SignedLogRoot
+	}{
+		{root: trillian.SignedLogRoot{TimestampNanos: 2267709, RootHash: []byte("Islington"), TreeSize: 2}},
+	} {
+		sig, err := signer.SignLogRoot(&test.root)
+		if err != nil {
+			t.Errorf("Failed to sign log root: %v", err)
+		}
+		if got := len(sig.Signature); got == 0 {
+			t.Errorf("len(sig): %v, want > 0", got)
+		}
+		if got, want := sig.HashAlgorithm, sigpb.DigitallySigned_SHA256; got != want {
+			t.Errorf("Hash alg incorrect, got %v expected %d", got, want)
```

Comment:
nit: Use "%s" for the enums, so we get names instead of numbers?

Reply:
Done, and applied elsewhere too.